Userland/Libraries/LibJS/Parser.cpp

```diff
@@ -4135,8 +4135,8 @@ void Parser::expected(char const* what)
 {
     auto message = m_state.current_token.message();
     if (message.is_empty())
-        message = DeprecatedString::formatted("Unexpected token {}. Expected {}", m_state.current_token.name(), what);
-    syntax_error(message);
+        message = String::formatted("Unexpected token {}. Expected {}", m_state.current_token.name(), what).release_value_but_fixme_should_propagate_errors();
```

Comment:
How is this related to SourceCode?

Reply:
It seems less intrusive to use ` auto message = m_state.current_token.message().to_deprecated_string();` here instead, so I've updated this change.